lib/nxt_pipeline/step.rb

```diff
@@ -54,6 +55,14 @@ def execute(**changeset)
 
     attr_writer :result, :status, :error, :mapped_options
     attr_reader :constructor, :options_mapper
+    
+    def instrumentalize_callable(callable, args)
+      if args.last.is_a?(Hash)
+        callable.call(*args.take(args.length - 1), **args.last)
```

Comment:
@lutfidemirci Regarding our discussion that this might not work if `args` only contains one object which is a `Hash`, I now know why it still works: `args.length - 1` equals `0` and `Array#take(0)` simply returns an empty array `[]`. Passing an empty array with split operator is fine.